src/datatablerow/utils.jl

```diff
@@ -0,0 +1,191 @@
+                        eq = false # miss
+                        break
+                    end
+                end
+                if eq # hit
+                    gix = groups !== nothing ? groups[g_row] : 0
+                    break
+                end
+            end
+            slotix = slotix & szm1 + 1 # check the next slot
+            probe += 1
+            @assert probe < sz
+        end
+        if groups !== nothing
+            groups[i] = gix
+        end
+    end
+    return ngroups, rhashes, gslots
+end
+
+# Builds RowGroupDict for a given datatable.
+# Partly uses the code of Wes McKinney's groupsort_indexer in pandas (file: src/groupby.pyx).
+function group_rows(dt::AbstractDataTable)
+    groups = Vector{Int}(nrow(dt))
+    ngroups, rhashes, gslots = row_group_slots(dt, groups)
+
+    # count elements in each group
+    stops = zeros(Int, ngroups)
+    @inbounds for g_ix in groups
+        stops[g_ix] += 1
+    end
+
+    # group start positions in a sorted table
+    starts = Vector{Int}(ngroups)
+    if !isempty(starts)
+        starts[1] = 1
+        @inbounds for i in 1:(ngroups-1)
+            starts[i+1] = starts[i] + stops[i]
+        end
+    end
+
+    # define row permutation that sorts them into groups
+    rperm = Vector{Int}(length(groups))
+    copy!(stops, starts)
+    @inbounds for (i, gix) in enumerate(groups)
+        rperm[stops[gix]] = i
+        stops[gix] += 1
+    end
+    stops .-= 1
+    return RowGroupDict(dt, ngroups, rhashes, gslots, groups, rperm, starts, stops)
+end
+
+# Find index of a row in gd that matches given row by content, 0 if not found
+function findrow(gd::RowGroupDict, dt::DataTable, row::Int)
+    (gd.dt === dt) && return row # same table, return itself
+    # different tables, content matching required
+    rhash = rowhash(dt, row)
+    szm1 = length(gd.gslots)-1
+    slotix = ini_slotix = rhash & szm1 + 1
+    while true
+        g_row = gd.gslots[slotix]
+        if g_row == 0 || # not found
+            (rhash == gd.rhashes[g_row] &&
+            isequal_row(gd.dt, g_row, dt, row)) # found
+            return g_row
+        end
+        slotix = (slotix & szm1) + 1 # miss, try the next slot
+        (slotix == ini_slotix) && break
+    end
+    return 0 # not found
+end
+
+# Find indices of rows in 'gd' that match given row by content.
+# return empty set if no row matches
+function Base.get(gd::RowGroupDict, dt::DataTable, row::Int)
+    g_row = findrow(gd, dt, row)
+    (g_row == 0) && return view(gd.rperm, 0:-1)
+    gix = gd.groups[g_row]
+    return view(gd.rperm, gd.starts[gix]:gd.stops[gix])
+end
+
+function Base.getindex(gd::RowGroupDict, dtr::DataTableRow)
+    g_row = findrow(gd, dtr.dt, dtr.row)
+    (g_row == 0) && throw(KeyError(dtr))
+    gix = gd.groups[g_row]
+    return view(gd.rperm, gd.starts[gix]:gd.stops[gix])
+end
+
+# Check if there is matching row in gd
+Base.in(gd::RowGroupDict, dt::DataTable, row::Int) = (findrow(gd, dt, row) != 0)
```

Comment:
I don't think it's a good idea to overload `in`, as other definitions take two arguments. Better define a custom function with a more explicit name. Or even better, use `findrow(...) != 0` everywhere.